java/feathub-connectors/flink-connectors/flink-connector-redis/src/main/java/com/alibaba/feathub/flink/connectors/redis/sink/RedisSinkFunction.java

```diff
@@ -0,0 +1,252 @@
+    private final String password;
+    private final int dbNum;
+
+    private final String tableName;
+    private final List<Integer> keyFieldIndices;
+    private final int timestampFieldIndex;
+
+    private final String evalScript;
+    private String evalScriptSHA;
+
+    private transient Jedis jedis;
+
+    public RedisSinkFunction(
+            ReadableConfig config,
+            ResolvedSchema schema,
+            DynamicTableSink.DataStructureConverter converter) {
+        this.converter = converter;
+        this.host = config.get(HOST);
+        this.port = config.get(PORT);
+        this.password = config.get(PASSWORD);
+        this.dbNum = config.get(DB_NUM);
+        this.tableName = config.get(TABLE_NAME);
+        this.keyFieldIndices = checkAndGetKeyFieldIndices(config, schema);
+        this.timestampFieldIndex = checkAndGetTimestampFieldIndex(config, schema);
+        this.evalScript = getEvalScript(timestampFieldIndex);
+
+        validateConfigAndSchema(config, schema, keyFieldIndices, timestampFieldIndex);
+    }
+
+    @Override
+    public void open(Configuration parameters) throws Exception {
+        super.open(parameters);
+        jedis = new Jedis(host, port);
+        if (!StringUtils.isNullOrWhitespaceOnly(password)) {
+            jedis.auth(password);
+        }
+        jedis.select(dbNum);
+        if (timestampFieldIndex >= 0) {
+            evalScriptSHA = jedis.scriptLoad(this.evalScript);
+        }
+    }
+
+    @Override
+    public void invoke(RowData data, Context context) {
+        Row row = (Row) converter.toExternal(data);
+        if (row == null) {
+            Preconditions.checkArgument(
+                    data == null, "RowData %s cannot be converted to Row", data);
+            return;
+        }
+
+        List<String> keys = new ArrayList<>();
+        for (int i : keyFieldIndices) {
+            keys.add(String.valueOf(row.getField(i)));
+        }
+        String key = tableName + ":" + String.join("#", keys);
+
+        Map<String, String> value = new HashMap<>();
+        int arity = data.getArity();
+        int keyFieldIndex = 0;
+        for (int i = 0; i < arity; i++) {
+            if (timestampFieldIndex == i) {
+                // timestamp values have specific serialization strategy
+                continue;
+            }
+
+            if (keyFieldIndex < keyFieldIndices.size() && keyFieldIndices.get(keyFieldIndex) == i) {
+                // don't store key values
+                keyFieldIndex++;
+                continue;
+            }
+
+            Object fieldValue = row.getField(i);
+
+            if (fieldValue == null) {
+                // don't store null values
+                continue;
+            }
+
+            value.put(String.valueOf(i), String.valueOf(fieldValue));
+        }
+
+        if (timestampFieldIndex < 0) {
+            if (value.isEmpty()) {
+                return;
+            }
+            this.jedis.hmset(key, value);
```

Comment:
Since hmset is deprecated, can we use `hset` with multiple field-value pairs?

https://redis.io/commands/hmset/